app/models/broker_agency_profile.rb

```diff
@@ -3,17 +3,17 @@ class BrokerAgencyProfile
   include SetCurrentUser
   include Mongoid::Timestamps
   include AASM
+  include IndividualMarketBehaviors
 
   embedded_in :organization
 
-  MARKET_KINDS = %W[individual shop both]
-
-  MARKET_KINDS_OPTIONS = {
+  MARKET_KINDS = individual_market_is_enabled? ? %W[individual shop both] : %W[shop]
```

Comment:
Some things to note here

MARKET_KINDS needs to be adjusted based on the settings also - it's used in validations.

I don't want to use the Settings object directly in the models - I've defined a concern that has a method that handles the logic and you can use that in each model as necessary. 

This has several advantages - one it encapsulates the actual implementation of the logic so we can change it in a single place in the future. 

Two it gives us an idiomatic method name to use instead - a future developer can clearly see what the intend is by reading this line. 

Three it gives us a pathway to define this in a separate gem at a later date